### PR TITLE
Drupal 7: Implement values array casting in __call() magic method

### PR DESCRIPTION
**Pull request by ademarco**
This pull request aim to get rid of explicit array-casting in Drupal 7 field handlers in favor of a casting happening in handler abstract class's __call() magic method. It also refactors Drupal 7 field handlers in order to allow object mocking. A basic PHPUnit test coverage with partial mocking by http://docs.mockery.io/ is also provided.

### CODE REVIEW

File: tests/Drupal/Tests/Driver/Drupal7FieldHandlerTest.php [hunk 1]
```diff
@@ -0,0 +1,87 @@
...
+   * Data provider.
+   *
+   * @return array
+   */
+  public function dataProvider()
+  {
+    return [
+
+      // Test default text field provided as simple text.
+      [
```
Comment (jhedstrom) [inline tests/Drupal/Tests/Driver/Drupal7FieldHandlerTest.php]: For now, I'd like to maintain PHP 5.3 compatibility if we can--Drupal 7 still supports that.

**Comment by ademarco** [comment]
Sure, here we go!

**Comment by ademarco** [comment]
Any other changes you wish me to have a look at?

**Comment by jhedstrom** [comment]
Sorry for the delay. Thanks!

**Comment by ademarco** [comment]
No problem at all, thanks for the merge!

On Mon, Mar 23, 2015 at 7:47 PM, Jonathan Hedstrom <notifications@github.com

**Pull request merged**